src/devices/Hx711/HX711.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device;
+using System.Device.Gpio;
+using System.Linq;
+using UnitsNet;
+
+namespace Iot.Device.Hx711
+{
+    /// <summary>
+    /// Hx711 - Weight scale Module
+    /// </summary>
+    /// <remarks>DataSheet: https://html.alldatasheet.com/html-pdf/1132222/AVIA/Hx711/109/1/Hx711.html</remarks>
+    public sealed class Hx711 : IDisposable
+    {
+        private readonly GpioController _gpioController;
+        private readonly bool _shouldDispose;
+
+        private readonly int _pinDout;
+        private readonly int _pinPdSck;
+
+        private readonly Hx711Options _options;
+        private readonly object _readLock;
+        private readonly Hx711Reader _reader;
+
+        private readonly List<double> _referenceUnitList = new();
+
+        private bool _isInitialize;
+        private bool _isCalibrated;
+        private int _tareValue;
+
+        /// <summary>
+        /// Offset value from 0 at startup
+        /// </summary>
+        private int _offsetFormZero;
+
+        /// <summary>
+        /// Conversion ratio between Hx711 units and grams
+        /// </summary>
+        public double ReferanceUnit { get; private set; }

```

Comment:
should this actually be public?

Reply:
If you don't leave it public, you can only know value of `ReferenceUnit` if you have source files, during debugging and after the calibration process.
Instead in this way it is possible to be able to read the value, after the calibration process, even if it is a Nuget package.
Don't you think that's correct?
P.s.
I have now seen grammar error Refer`a`nce Unit instead of Refer`e`nceUnit. I fix it.

Reply:
I'm more confused about the name of the property, actually. An "unit" is something like "meters" or "kilograms". I would call this somethink like `CalibrationValue` or `MeasurementValuePerGram` or so. I do agree that it should be public, though.

Reply:
I'll suggest to use Factor or Ratio in the name somewhere. Comment says "conversion ratio" - that sounds like fair name to me.

Reply:
@krwq I changed name of this field from ReferanceUnit to ConversionRatio. Need more changes to close the pr?